watchtower/blob/justice_kit.go

```diff
@@ -0,0 +1,406 @@
+	ErrKeySize = fmt.Errorf(
+		"chacha20poly1305 key size must be %d bytes", KeySize,
+	)
+)
+
+// PubKey is a 33-byte, serialized compressed public key.
+type PubKey [33]byte
+
+// JusticeKit is lé Blob of Justice. The JusticeKit contains information
+// required to construct a justice transaction, that sweeps a remote party's
+// revoked commitment transaction. It supports encryption and decryption using
+// chacha20poly1305, allowing the client to encrypt the contents of the blob,
+// and for a watchtower to later decrypt if action must be taken. The encoding
+// format is versioned to allow future extensions.
+type JusticeKit struct {
+	// SweepAddress is the p2wkh address of the output where the client's
+	// fund will be deposited. This value is included in the blobs, as
+	// opposed to the session info, such that the sweep addresses can't be
+	// correlated across sessions and/or towers.
+	SweepAddress [34]byte
+
+	// RevocationPubKey is the compressed pubkey that guards the revocation
+	// clause of the remote party's to-local output.
+	RevocationPubKey PubKey
+
+	// LocalDelayPubKey is the compressed pubkey in the to-local script of
+	// the remote party, which guards the path where the remote party
+	// claims their commitment output.
+	LocalDelayPubKey PubKey
+
+	// CSVDelay is the relative timelock in the remote party's to-local
+	// output, which the remote party must wait out before sweeping their
+	// commitment output.
+	CSVDelay uint32
+
+	// CommitToLocalSig is a signature under RevocationPubKey using
+	// SIGHASH_ALL.
+	CommitToLocalSig lnwire.Sig
+
+	// P2WKHPubKey is the public key in the to-remote output of the revoked
+	// commitment transaction.
+	//
+	// NOTE: This value is only used if it contains a valid compressed
+	// public key.
+	P2WKHPubKey PubKey
+
+	// P2WKHSig is a signature under P2WKHPubKey using SIGHASH_ALL.
+	//
+	// NOTE: This value is only used if P2WKHPubKey contains a valid
+	// compressed public key.
+	P2WKHSig lnwire.Sig
+}
+
+// CommitToLocalScript returns the serialized witness script for the commitment
+// to-local output.
+func (b *JusticeKit) CommitToLocalScript() ([]byte, error) {
+	revocationPubKey, err := btcec.ParsePubKey(
+		b.RevocationPubKey[:], btcec.S256(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	localDelayedPubKey, err := btcec.ParsePubKey(
+		b.LocalDelayPubKey[:], btcec.S256(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return lnwallet.CommitScriptToSelf(
+		b.CSVDelay, localDelayedPubKey, revocationPubKey,
+	)
+}
+
+// HasP2WKHOutput returns true if the blob contains a p2wkh pubkey and sig.
+func (b *JusticeKit) HasP2WKHOutput() bool {
+	return btcec.IsCompressedPubKey(b.P2WKHPubKey[:])
+}
+
+// CommitP2WKHScript returns the serialized pubkey script for the commitment
+// to-remote output.
+func (b *JusticeKit) CommitP2WKHScript() ([]byte, error) {
+	if btcec.IsCompressedPubKey(b.P2WKHPubKey[:]) {
+		return nil, nil
```

Comment:
Looks like this should have a `!`, and also return an error in the failing case (so not a compressed key).